packages/clients/textLocator/src/plugins/GeometrySearch/store/actions/setupDrawReaction.ts

```diff
@@ -0,0 +1,35 @@
+import { PolarActionHandler, PolarStore } from '@polar/lib-custom-types'
+import debounce from 'lodash.debounce'
+import { Feature } from 'ol'
+import VectorSource, { VectorSourceEvent } from 'ol/source/Vector'
+import { GeometrySearchGetters, GeometrySearchState } from '../../types'
+
+let debouncedSearchGeometry: PolarActionHandler<
+  GeometrySearchState,
+  GeometrySearchGetters
+>
+
+export function setupDrawReaction(
+  this: PolarStore<GeometrySearchState, GeometrySearchGetters>,
+  { rootGetters, dispatch }
+) {
+  // features added multiple times; avoid overly requesting
+  debouncedSearchGeometry = debounce(
+    (feature) => dispatch('searchGeometry', feature),
+    20
+  ).bind(this)
+
+  // only keep a single feature in the draw tool
+  const drawSource = rootGetters['plugin/draw/drawSource'] as VectorSource
```

Comment:
I think we have to improve upon our typing here. In an ideal world, there would be no need for the type assertion 🎩